### PR TITLE
Exporting as ESM and CJS

### PR DESCRIPTION
**Pull request by anthonyalayo**
1. don't bundle modules, that allows for more effective tree shaking
2. remove the browser export, it seems unnecessary in 2023
3. keep exporting cjs as a single file
4. move emotion into peer dependencies, which is recommended by the project

This presently does not build, but it's almost there. I'll need a bit of help @pionxzh @rtritto

### CODE REVIEW

**Comment by pionxzh** [comment]
> Should we convert this project to ESM insead of convert it for export? ESM Guide: [Pure ESM package](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c)
> 
> > 4. move emotion into peer dependencies, which is recommended by the project
> 
> I think it's better to split this in a separated PR

yes, let's move it to another one.

**Comment by pionxzh** [comment]
Let me check what's wrong. 👌  Sorry I was so busy.

File: rollup.config.ts [hunk 1]
```diff
@@ -115,7 +109,7 @@ const dtsMatrix = (): RollupOptions[] => {
     cache,
     output: {
       file: resolve(outputDir, `${output}.d.ts`),
-      format: 'es'
+      format: 'umd'
```
Comment (pionxzh) [inline rollup.config.ts:112]: should we just use `cjs` instead?

**Comment by rtritto** [comment]
Needed this (changes)[https://github.com/TexteaInc/json-viewer/pull/181#discussion_r1072587828] too

**Comment by pionxzh** [comment]
> Needed these changes [#181 (comment)](https://github.com/TexteaInc/json-viewer/pull/181#discussion_r1072587828) too

Not sure if it is just me or what. I couldn't see the content of your comment 😵‍💫

**Comment by rtritto** [comment]
Just check the review comments

**Comment by pionxzh** [comment]
@anthonyalayo Sorry that we did a rebasing to remove some sensitive content from the history.
And now the history is messed up 😢 , I think we can close this first as most of the content has been split to other PRs. And we don't have a plan to drop the browser build. Is that ok?

**Comment by anthonyalayo** [comment]
Sounds good!

**Pull request closed without merging**